kayenta-standalone-canary-analysis/src/main/java/com/netflix/kayenta/standalonecanaryanalysis/service/CanaryAnalysisService.java

```diff
@@ -100,10 +113,40 @@ public CanaryAnalysisExecutionResponse initiateCanaryAnalysisExecution(
   }
 
   public CanaryAnalysisExecutionStatusResponse getCanaryAnalysisExecution(
-      String canaryAnalysisExecutionId) {
-    Execution execution =
-        executionRepository.retrieve(Execution.ExecutionType.PIPELINE, canaryAnalysisExecutionId);
-    return fromExecution(execution);
+      String canaryAnalysisExecutionId, String nullableStorageAccountName) {
+
+    try {
+      Execution execution =
+          executionRepository.retrieve(Execution.ExecutionType.PIPELINE, canaryAnalysisExecutionId);
+      return fromExecution(execution);
+    } catch (ExecutionNotFoundException e) {
+      return Optional.ofNullable(nullableStorageAccountName)
+          .map(
+              storageAccountName -> {
+                String resolvedStorageAccountName =
+                    CredentialsHelper.resolveAccountByNameOrType(
+                        storageAccountName,
+                        AccountCredentials.Type.OBJECT_STORE,
+                        accountCredentialsRepository);
+
+                StorageService storageService =
+                    storageServiceRepository
+                        .getOne(resolvedStorageAccountName)
+                        .orElseThrow(
+                            () ->
+                                new IllegalArgumentException(
+                                    String.format(
+                                        "No storage service was configured for %s unable to load archived results.",
+                                        resolvedStorageAccountName)));
+
+                return (CanaryAnalysisExecutionStatusResponse)
+                    storageService.loadObject(
+                        resolvedStorageAccountName,
+                        ObjectType.STANDALONE_CANARY_RESULT_ARCHIVE,
+                        canaryAnalysisExecutionId);
+              })
+          .orElseThrow(() -> e);
```

Comment:
@marchello2000 What do you think about that?

It throws the original error if you didn't add ?storageAccount=xxxx to the request.

It throws an Illegal Argument Exception or other downstream exceptions from `storageService.loadObject` if you supply that new optional query param and it doesn't load.

Reply:
👍